src/middlewared/middlewared/plugins/pool_/attach_disk.py

```diff
@@ -44,52 +44,52 @@ async def attach(self, job, oid, options):
         else:
             verrors.add('pool_attach.target_vdev', 'Unable to locate VDEV')
             verrors.check()
+
         if topology_type in ('cache', 'spares'):
             verrors.add('pool_attach.target_vdev', f'Attaching disks to {topology_type} not allowed.')
         elif topology_type == 'data':
             # We would like to make sure here that we don't have inconsistent vdev types across data
             if vdev['type'] not in ('DISK', 'MIRROR'):
                 verrors.add('pool_attach.target_vdev', f'Attaching disk to {vdev["type"]} vdev is not allowed.')
 
-        if osc.IS_FREEBSD and pool['encrypt'] == 2:
+        if pool['encrypt'] == 2:
             if not options.get('passphrase'):
                 verrors.add('pool_attach.passphrase', 'Passphrase is required for encrypted pool.')
             elif not await self.middleware.call('disk.geli_testkey', pool, options['passphrase']):
                 verrors.add('pool_attach.passphrase', 'Passphrase is not valid.')
 
-        if osc.IS_LINUX and options.get('passphrase'):
-            verrors.add(
-                'pool_attach.passphrase',
-                'This field is not valid on this platform.'
-            )
-
         # Let's validate new disk now
         await self.middleware.call('disk.check_disks_availability', verrors, [options['new_disk']], 'pool_attach')
         verrors.check()
 
-        guid = vdev['guid'] if vdev['type'] == 'DISK' else vdev['children'][0]['guid']
-        disks = {options['new_disk']: {'create_swap': topology_type == 'data', 'vdev': []}}
-        enc_disks = await self.middleware.call(
-            'pool.format_disks', job, disks,
-            {'enc_keypath': pool['encryptkey_path'], 'passphrase': options.get('passphrase')}
-        )
+        disks = {options['new_disk']: {'create_swap': topology_type == 'data'}}
+        await self.middleware.call('pool.format_disks', job, disks)
+        await self.middleware.call('geom.cache.invalidate')
```

Comment:
Do we perhaps want to invalidate cache at the end of `pool.format_disks` instead of making sure we do it each time after calling it ?

Reply:
great idea. I'll make that change